File: c/hamming/hamming.c
#include "hamming.h"
#include <stdio.h>

int compute(const char *lhs, const char *rhs) {
  printf("lhs: %s, rhs: %s\n", lhs, rhs);

  if (!lhs || !rhs) {
    return -1;
  }

  int distance = 0;
  for (; *lhs || *rhs; lhs++, rhs++) {
    if (!*lhs || !*rhs) {
      return -1;
    }

    if (*lhs != *rhs) {
      distance++;
    }
  }

  return distance;
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
	if #a ~= #b then
		return -1
	end

	local result = 0

	for i = 1, #a do
		if a:sub(i, i) ~= b:sub(i, i) then
			result = result + 1
		end
	end

	return result
end

return Hamming
